doc/guides/networking/net-stack-architecture.rst

```diff
@@ -0,0 +1,137 @@
+.. _network_stack_architecture:
+
+Network Stack Architecture
+##########################
+
+High level overview of the network stack
+****************************************
+
+.. figure:: zephyr_netstack_overview.svg
+    :align: left
+    :alt: Overview of the network stack architecture
+    :figclass: align-center
+
+    Network stack overview
+
+*The network stack is layered and it consists of the following parts:*
+
+* **Network Application.** The network application uses
+  :ref:`BSD socket API <bsd_sockets_interface>` to create a network connection,
+  send or receive data, and close a connection. The application can also use
+  the :ref:`network management API <net_mgmt_interface>` to set network
```

Comment:
> to set network  related parameters

Suggested: "to configure network and set related parameters".

Kinda make more explicit that we offer both data-plane and control-plane APIs, and application has (or can have) full control of networking.